### DISCUSSION TITLE
How to disable hydration on select pages?

### DISCUSSION THREAD

**Discussion by clouedoc** [nuxt/nuxt#17269] [Questions]
Hello,

I am building a website generated from a PostgreSQL database.
I do not want to neither build an API nor host a server, so I opted for pre-rendering and static generation.

The problem is that, in production, the right page appears for a brief amount of time, and then disappears.
Everything goes fine in development.
I believe that the app asks the API for the latest data, which is a behaviour that I am not interested in.

Here is a reproduction link: https://62af06f77e4d5037ba5e977c--avis-mandataire.netlify.app/

Here is the relevant code:

```vue
<template>
  <div>
    <h1>Accueil</h1>

    <p>
      Bienvenue sur avis-mandataire.com, le site de référencement des
      mandataires préféré des français.
    </p>

    <h2>Mandataires par département</h2>

    <ul>
      <li v-for="departement in departements" :key="departement.department">
        <a :href="'/departement/' + departement">{{ departement }}</a>
      </li>
    </ul>
  </div>
</template>

<script setup>
const departements = await $fetch('/api/departements')
</script>
```

Here is my `nuxt.config.js`:

```js
import { defineNuxtConfig } from "nuxt";

// https://v3.nuxtjs.org/api/configuration/nuxt.config
export default defineNuxtConfig({
  typescript: {
    shim: false,
  },
  modules: ["@nuxtjs/tailwindcss"],
  ssr: true,
  target: "static",
});
```

## My research

I have found this relevant issue: https://github.com/nuxt/framework/issues/648 and this relevent PR https://github.com/nuxt/framework/pull/4390

## Workarounds

- Exposing an API to my PostgreSQL database using serverless functions (I loose the advantage of relying on the CDN 100%)
- `rm -rf .output/public/_nuxt/*.mjs` (I'm definitely going to hell)

Thank you for your attention

**Comment by TheAlexLichter** [comment]
Your problem is not the hydration itself but the missing support of "full-static", where the state of the server is added as JSON (or similar) and will replace API calls with fetching the saved state.

If it is only this page, you could already save the result of the API call as JSON via Nuxt hooks and then import the JSON in your component.

**Reply by clouedoc** [reply]
Thank for your insight.

edit: I asked you some clarifications but I misread your answers. I'll do some testing with your workaround before asking again!

**Reply by TotomInc** [reply]
Please report back if you found something, I would really love to have the same behaviour in my Nuxt v3 websites.

I wanted to do fully static SSG, without needing a back-end server during client-side navigation, which is something that we _unsurprisingly_ cannot do easily in Nuxt 3.

I already wrote a discussion thread here about it but haven't found a way (I instead migrated to Next.js which supports SSG _very_ well :sweat_smile:): https://github.com/nuxt/framework/discussions/4741

**Reply by kissu** [reply]
@TotomInc the migration from Vue2 -> Vue3 takes some time and it's not that easy I guess.
There are indeed various frameworks that support a (full) static SSG right now, this wasn't the prio of the team I guess.
